src/app/components/shared/date-time-filter/filter.component.ts

```diff
@@ -0,0 +1,199 @@
+
+    // if the new filters will be the same as the old filters, do not emit a filter update event
+    if (newFilters && this.model) {
+      this.constructedFilters.emit(newFilters);
+    }
+  }
+
+  public isInvalidDate(date: string | NgbDate): boolean {
+    if (!date) {
+      return false;
+    } else if (typeof date === "string") {
+      return true;
+    }
+
+    const isValidDate = this.calendar.isValid(date)
+    return date?.year.toString().length < 4 || !isValidDate;
+  }
+
+  /**
+   * Validates that the start date comes before the end date
+   *
+   * @param formStartDate A date object of type `NgbDate` supplied by the bootstrap date input representing the start date
+   * @param formEndDate A date object of type `NgbDate` supplied by the bootstrap date input representing the end date
+   * @returns Whether the start date comes before the end date, or the two dates are equal
+   */
+  public dateIsInBoundary(formStartDate: NgbDate, formEndDate: NgbDate): boolean {
+    // if the user has not set a start date and end date, we know that the date range is not defined and therefore, cannot be out of bounds
+    if (!isInstantiated(formStartDate) || !isInstantiated(formEndDate)) {
+      return true;
+    }
+
+    // converting to a date object allows the comparison of dates using boolean operators
+    const startDate = new Date(formStartDate.year, formStartDate.month, formStartDate.day);
+    const endDate = new Date(formEndDate.year, formEndDate.month, formEndDate.day);
+
+    return startDate <= endDate;
+  }
+
+  private generateFilters(model: Model): Filters | null {
+    const filter: InnerFilter = {};
+
+    if (this.site) {
+      filter["sites.id"] = { eq: this.site.id };
+    } else if (this.region) {
+      filter["regions.id"] = { eq: this.region.id };
+    } else if (this.project) {
+      filter["projects.id"] = { eq: this.project.id };
+    }
+
+    this.setDateFilter(filter, model);
+    this.setTimeOfDayFilter(filter, model);
+
+    if (JSON.stringify(filter) === JSON.stringify({})) {
+      return null;
+    }
+
+    if (JSON.stringify({ filter }) !== this.previousFilters) {
+      this.previousFilters = JSON.stringify({filter });
+      return { filter };
+    }
+
+    return null;
+  }
+
+  private setDateFilter(
+    filter: InnerFilter,
+    model: Model
+  ): void {
+    const filterStartDate = model?.dateStartedAfter;
+    const filterEndDate = model?.dateFinishedBefore;
+
+    // if there is no dates specified or the date boundaries are not correct, bail out
+    if (
+      !model?.dateFiltering ||
+      (!filterStartDate && !filterEndDate) ||
+      !this.dateIsInBoundary(filterStartDate, filterEndDate)
+    ) {
+      return;
+    }
+
+    if (!this.isInvalidDate(model.dateStartedAfter) && model.dateStartedAfter) {
+      filter["recordedDate"] ??= {};
+      filter["recordedDate"].greaterThanOrEqual =
+        `${this.formatter.format(model.dateStartedAfter)}T00:00:00.000Z`;
```

Comment:
This method currently uses the NgBootstrap date formatter (it's not in the scope of the method)

However, adding the time to the date was a hacky trick that is no longer needed since we are no longer mixing date and time filter objects together (yay!) as a way to reverse the hacky `T23:59:59.000Z` concatenation seen in a comment earlier